integration/inspektor-gadget/trace_dns_test.go

```diff
@@ -127,20 +156,37 @@ func TestTraceDns(t *testing.T) {
 				e.Runtime.ContainerID = ""
 			}
 
-			return ExpectEntriesToMatch(output, normalize, expectedEntries...)
-		},
-	}
+			compFn := func(a, b any) bool {
+				x := a.(*tracednsTypes.Event)
+				y := b.(*tracednsTypes.Event)
+				return x.Comm == y.Comm &&
+					x.Qr == y.Qr &&
+					x.Nameserver == y.Nameserver &&
+					x.PktType == y.PktType &&
+					x.DNSName == y.DNSName &&
+					x.QType == y.QType &&
+					x.Rcode == y.Rcode &&
+					x.Latency == y.Latency &&
+					x.NumAnswers == y.NumAnswers &&
+					reflect.DeepEqual(x.Addresses, y.Addresses) &&
+					x.Uid == y.Uid &&
+					x.Gid == y.Gid &&
+					x.Protocol == y.Protocol &&
+					((x.Qr == tracednsTypes.DNSPktTypeQuery &&
+						x.DstPort == y.DstPort && x.SrcIP == y.SrcIP) ||
+						(x.Qr == tracednsTypes.DNSPktTypeResponse &&
+							x.SrcPort == y.SrcPort &&
+							x.DstIP == y.DstIP))
+			}
```

Comment:
Under the default approach, we are matching all the fields generated by gadget event. I think for the DNS gadget event, we are interested in a subset of the fields (like the destination port for query should be 53) and not necessarily all fields (we don't care so much about the source port of the bustbox pod running the `nslookup`). Maybe this could be beneficial for other gadget tests going forward? We maybe interested to test only the fields we can control during test and test them?

Reply:
I agree with Francis, the `normalize` function should do the trick you're looking for.